diff --git a/ELF/LinkerScript.cpp b/ELF/LinkerScript.cpp
index 8bdbd8db2..e193296db 100644
--- a/ELF/LinkerScript.cpp
+++ b/ELF/LinkerScript.cpp
@@ -1171,7 +1171,7 @@ template <class ELFT> void OutputSectionCommand::writeTo(uint8_t *Buf) {
   if (Filler)
     fill(Buf, Sections.empty() ? Sec->Size : Sections[0]->OutSecOff, Filler);
 
-  parallelForEachN(0, Sections.size(), [=](size_t I) {
+  parallelForEachN(0, Sections.size(), [=,&Sections](size_t I) {
     InputSection *IS = Sections[I];
     IS->writeTo<ELFT>(Buf);
 
