diff --git a/contrib/llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp b/contrib/llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp
index 3664484..a9f5acb 100644
--- a/contrib/llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp
+++ b/contrib/llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp
@@ -154,6 +154,15 @@ Instruction *InstCombiner::foldSelectOpOp(SelectInst &SI, Instruction *TI,
       return nullptr;
     }
 
+    // Prevent folding if the select matches a min/max pattern
+    {
+      Value *left, *right;
+      Instruction::CastOps castOp;
+      auto selectPattern = matchSelectPattern(&SI, left, right, &castOp);
+      if (SelectPatternResult::isMinOrMax(selectPattern.Flavor))
+        return nullptr;
+    }
+
     // Fold this by inserting a select from the input values.
     Value *NewSI =
         Builder->CreateSelect(SI.getCondition(), TI->getOperand(0),
