diff --git a/cmake/anydsl_runtime-config.cmake.in b/cmake/anydsl_runtime-config.cmake.in
index 4cb689a..24c87f2 100644
--- a/cmake/anydsl_runtime-config.cmake.in
+++ b/cmake/anydsl_runtime-config.cmake.in
@@ -256,7 +256,7 @@ function(anydsl_runtime_wrap outfiles)
         set(HLS_COMMAND COMMAND ${CMAKE_COMMAND} -D_basename=${_basename} -DHLS_FLAGS=${HLS_FLAGS} -P ${AnyDSL_runtime_DIR}/build_xilinx_hls.cmake)
     endif()
 
-    set(_basepath ${CMAKE_CURRENT_BINARY_DIR}/${CMAKE_CFG_INTDIR}/${_basename})
+    set(_basepath ${CMAKE_CURRENT_BINARY_DIR}/$<CONFIG>/${_basename})
     set(_llfile ${_basepath}.ll)
     set(_cfile ${_basepath}.c)
     set(_objfile ${_basepath}.o)
