index 0000000..0000000 100644
--- a/llvm/include/llvm/Support/type_traits.h
+++ b/llvm/include/llvm/Support/type_traits.h
@@ -177,7 +177,7 @@ class is_trivially_copyable {
       (has_deleted_copy_assign || has_trivial_copy_assign) &&
       (has_deleted_copy_constructor || has_trivial_copy_constructor);
 
-#ifdef HAVE_STD_IS_TRIVIALLY_COPYABLE
+#if defined(HAVE_STD_IS_TRIVIALLY_COPYABLE) && (!defined(_MSC_VER) || (_MSC_VER <= 1926))
   static_assert(value == std::is_trivially_copyable<T>::value,
                 "inconsistent behavior between llvm:: and std:: implementation of is_trivially_copyable");
 #endif
